### PR TITLE
(EWL-1274): Freewheel: Ads custom control bar

### PR DESCRIPTION
**Pull request by psaint-j**
# About the type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Test plan

- [ ] Empty data
- [ ] Invalid data
- [ ] Correct data

# Checklist
### Before code review

- [x] Move Jira ticket into PR
- [x] Check with another developers that features sounds good

### After code review

- [ ] Move Jira ticket into QA

### After QA validation
If conflicts are present in github:
  - [ ] `git pull --rebase origin master`
  - [ ] Fix conflicts
  - [ ] `git rebase --continue` repeat until there is no more conflicts
  - [ ] `git push --force-with-lease`

If there's no conflicts
- [ ] squash and merge
- [ ] delete branch

### CODE REVIEW

File: src/components/JWPlayerETP/Controls/LiveStatus/index.js [hunk 1]
```diff
@@ -9,14 +9,20 @@ const LiveStatus = ({ rewindCounts, isLive, onSeek, seekMax }) => {
...
-      <BackToLiveAction rewindCounts={rewindCounts} onClick={onClickBackToLive}>
-        {rewindCounts && <ForwardIcon src={forward} alt="Forward" />}
-        <Icon isLive={isLive} rewindCounts={rewindCounts}>
-          <Text>LIVE</Text>
-        </Icon>
-      </BackToLiveAction>
+      {isLive && (
+        <BackToLiveAction rewindCounts={rewindCounts} onClick={onClickBackToLive}>
+          {rewindCounts && <ForwardIcon src={forward} alt="Forward" />}
+          <Icon isLive={isLive} rewindCounts={rewindCounts}>
```
[the remark this answers is not in the document]
Comment (MathieuDebit) [reply src/components/JWPlayerETP/Controls/LiveStatus/index.js]: So I guess simply this?

```suggestion
          <Icon isLive rewindCounts>
```

File: src/components/JWPlayerETP/Controls/LiveStatus/index.js [hunk 2]
```diff
@@ -9,14 +9,20 @@ const LiveStatus = ({ rewindCounts, isLive, onSeek, seekMax }) => {
...
+            <Text>LIVE</Text>
+          </Icon>
+        </BackToLiveAction>
+      )}
 
-      {rewindCounts && <RewindText length={rewindCounts?.length}>{rewindCounts}</RewindText>}
+      {rewindCounts && (
+        <RewindText length={rewindCounts?.length} isLive={isLive}>
```
Comment (Xb0X) [inline src/components/JWPlayerETP/Controls/LiveStatus/index.js]: same here :) 
`<RewindText length={rewindCounts?.length} isLive={isLive}>`
could be 
`<RewindText length={rewindCounts?.length} isLive>`

File: src/components/JWPlayerETP/PlayerSkin/index.test.js [hunk 3]
```diff
@@ -265,5 +266,18 @@ describe('Components|PlayerSkin', () => {
...
+      useFullscreen.mockReturnValue([false, onFullscreenChange]);
+      // Given
+      const props = createDefaultProps({ isPlaying: true, isAdPlaying: true });
+      wrapper = shallow(<PlayerSkin {...props} />);
+
+      // When
+      wrapper.find(Controls).prop('onFullscreenChange')();
+
+      // Expect
+      expect(wrapper.find(Controls).prop('title')).toEqual('title');
```
Comment (narajaon) [inline src/components/JWPlayerETP/PlayerSkin/index.test.js]: Here we are not checking if the title is displayed or not, we are just checking if the prop is correct which is not the same

File: src/components/JWPlayerETP/PlayerSkin/index.test.js [hunk 4]
```diff
@@ -265,5 +266,18 @@ describe('Components|PlayerSkin', () => {
       // Expect
       expect(wrapper.find(Controls).prop('title')).toBeUndefined();
```
Comment (narajaon) [inline src/components/JWPlayerETP/PlayerSkin/index.test.js]: Here we are not checking if the title is displayed or not, we are just checking if the prop is undefined which is not the same

File: src/components/JWPlayerETP/PlayerSkin/index.test.js [hunk 5]
```diff
@@ -244,26 +248,57 @@ describe('Components|PlayerSkin', () => {
...
+      );
+
+      // When
+      act(() => {
+        wrapper.find(Controls).prop('onFullscreenChange')();
+      });
+
+      // Expect
+      expect(wrapper.find(TitleContainer).text()).toEqual('');
+    });
```
Comment (JulienKode) [inline src/components/JWPlayerETP/PlayerSkin/index.test.js]: - Can we `unmount` the component here ?
- Does `act` can be used with enzyme ?

File: src/components/JWPlayerETP/Controls/index.js [hunk 6]
```diff
@@ -83,6 +85,8 @@ Controls.defaultProps = {
   isLive: false,
   rewindCounts: null,
   title: undefined,
+  // eslint-disable-next-line react/default-props-match-prop-types
+  isAdPlaying: false,
```
Comment (JulienKode) [inline src/components/JWPlayerETP/Controls/index.js]: Can you add it in the propTypes ?

Comment (MathieuDebit) [reply src/components/JWPlayerETP/Controls/index.js]: Actually it's technically in the propTypes via `...PlayerButtons.propTypes`. Maybe ESLint does not resolve it?

File: src/components/JWPlayerETP/PlayerSkin/index.test.js (hunk 5 above)
Comment (MathieuDebit) [reply src/components/JWPlayerETP/PlayerSkin/index.test.js]: Enzyme seems compatible with `act` yes

**Pull request merged**